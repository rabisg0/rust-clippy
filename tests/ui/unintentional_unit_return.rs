#![warn(clippy::unintentional_unit_return)]
#![feature(is_sorted)]

struct Struct {
    field: isize,
}

fn double(i: isize) -> isize {
    i * 2
}

/*
fn fn_with_closure<T, F, K>(mut v: Vec<T>, f: F) where
    F: FnMut(&T) -> K,
    K: Ord {
    v.sort_by_key(f)
}
*/

fn main() {
    let mut structs = vec![Struct { field: 2 }, Struct { field: 1 }];
    structs.sort_by_key(|s| {
        double(s.field);
    });
    structs.sort_by_key(|s| double(s.field));
    structs.is_sorted_by_key(|s| {
        double(s.field);
    });
}
