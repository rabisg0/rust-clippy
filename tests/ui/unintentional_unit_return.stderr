error: Semi-colon on the last line of this closure returns the unit type which also implements Ord.
  --> $DIR/unintentional_unit_return.rs:23:9
   |
LL |         double(s.field);
   |         ^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::unintentional-unit-return` implied by `-D warnings`

error: Semi-colon on the last line of this closure returns the unit type which also implements PartialOrd.
  --> $DIR/unintentional_unit_return.rs:27:9
   |
LL |         double(s.field);
   |         ^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

